### PR TITLE
Feature/dividends page presenter

### PR DESCRIPTION
**Pull request by Grsmto**
This PR includes:
- Dividends module activation page presenter
- Confirm/ModalTransactionQueue implementation

### CODE REVIEW

**Comment by Grsmto** [comment]
Still one typing error with the modal demos (saying `status` is required...). If you guys have an idea how to fix this.

**Comment by Grsmto** [comment]
Back to WIP until @monitz87 finishes fixes.

File: packages/new-polymath-ui/src/components/Grid/Grid.tsx [hunk 1]
```diff
@@ -9,16 +9,20 @@ import {
   gridAutoColumns,
   gridAutoRows,
   alignItems,
+  // @ts-ignore
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/Grid/Grid.tsx]: Why is this necessary?

File: packages/new-polymath-ui/src/components/Grid/docs/Grid.mdx [hunk 2]
```diff
@@ -23,31 +24,31 @@ In its normal form, `Grid` accepts basic css-grid properties.
...
 
-"Bootstrap-style" 12-columns grid is being used in the design. To achieve it you can use the `Grid.Row` component.
-The main difference with the classic Bootstrap grid is that **`Grid.Row` is a global container for your grid, not only a row** (so, then the name is not really accurate but could not find better).
+"Bootstrap-style" 12-columns grid is being used in the design. To achieve it you can use the `GridRow` component.
+The main difference with the classic Bootstrap grid is that **`GridRow` is a global container for your grid, not only a row** (so, then the name is not really accurate but could not find better).
 
 ### Important
 
-- — You can have defined any number of rows in a `Grid.Row`, just keep adding some `Grid.Col`.
+- — You can have defined any number of rows in a `GridRow`, just keep adding some `Grid.Col`.
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/Grid/docs/Grid.mdx]:
```suggestion
- — You can have defined any number of rows in a `GridRow`, just keep adding some `GridRow.Col`.
```

File: packages/new-polymath-ui/src/components/ModalConfirmTransactionQueue/ModalConfirmTransactionQueue.tsx [hunk 3]
```diff
@@ -1,20 +1,44 @@
...
 
-import { ModalConfirm, ModalConfirmProps } from '~/components/ModalConfirm';
+type ModalConfirmProps = typeHelpers.GetProps<typeof ModalConfirm>;
 
-type Props = ModalConfirmProps;
-export type ModalConfirmTransactionProps = JSX.LibraryManagedAttributes<
-  typeof ModalConfirmTransactionQueue,
-  Props
->;
+interface TransactionQueue extends types.TransactionQueueEntity {
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/ModalConfirmTransactionQueue/ModalConfirmTransactionQueue.tsx]: There is already a type for this in `new-shared` (`types.TransactionQueuePojo`)

File: packages/new-polymath-ui/src/components/ModalConfirmTransactionQueue/ModalConfirmTransactionQueue.tsx [hunk 4]
```diff
@@ -1,20 +1,44 @@
...
+  transactions: types.TransactionEntity[];
+}
 
-const ModalConfirmTransactionQueueBase = (props: Props) => {
-  return <ModalConfirm maxWidth={500} {...props} />;
-};
+interface Props extends ModalConfirmProps {
+  transactionQueue: TransactionQueue;
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/ModalConfirmTransactionQueue/ModalConfirmTransactionQueue.tsx]:
```suggestion
  transactionQueue: types.TransactionQueuePojo;
```

File: packages/new-polymath-ui/src/components/ModalConfirmTransactionQueue/TransactionItem/index.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import { IconCircled } from '~/components/IconCircled';
+import { Heading } from '~/components/Heading';
+import { CardPrimary } from '~/components/CardPrimary';
+import { Paragraph } from '~/components/Paragraph';
+import { getTransactionText } from '~/components/utils/contentMappings';
+
+import * as sc from './styles';
+
+interface TransactionItemProps {
+  transaction: types.TransactionEntity;
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/ModalConfirmTransactionQueue/TransactionItem/index.tsx]:
```suggestion
  transaction: types.TransactionEntityPojo;
```

Right now they are the same, but it might change in the future, and the Pojo suffixed types are the ones that represent the actual data the components would use, while the ones without the suffix are the normalized entities that we keep in the store

File: packages/new-polymath-ui/src/components/ModalTransactionQueue/ModalTransactionQueue.tsx [hunk 6]
```diff
@@ -18,28 +19,28 @@ const { TransactionQueueStatus } = types;
...
-    [ModalStatus.alert]: `An error occured with ${title}`,
-    [ModalStatus.success]: `${title} was successfully submitted`,
+    [ModalStatus.Idle]: `Proceed with ${title}`,
+    [ModalStatus.Loading]: `Proceed with ${title}`,
+    [ModalStatus.Warning]: `An error occured with ${title}`,
+    [ModalStatus.Alert]: `An error occured with ${title}`,
+    [ModalStatus.Success]: `${title} was successfully submitted`,
   }[status]);
 
 interface TransactionQueue extends types.TransactionQueueEntity {
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/ModalTransactionQueue/ModalTransactionQueue.tsx]: There is already a type for this in `new-shared` (`types.TransactionQueuePojo`)

Comment (monitz87) [reply packages/new-polymath-ui/src/components/ModalTransactionQueue/ModalTransactionQueue.tsx]: Also change its use in line 51 (I can't add a comment there for some reason)

File: packages/new-polymath-ui/src/components/ModalTransactionQueue/TransactionItem/TransactionItem.tsx [hunk 7]
```diff
@@ -24,27 +23,31 @@ interface TransactionItemProps {
 }
 
 const getIcon = (transaction: types.TransactionEntity) => {
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/ModalTransactionQueue/TransactionItem/TransactionItem.tsx]: Repeated use of `transaction.status` warrants defining `const { status } = transaction;` and using `status` accross the function.

File: packages/new-polymath-issuer/src/pages/SecurityTokensDividends/Container.tsx [hunk 8]
```diff
@@ -40,21 +40,27 @@ export class ContainerBase extends Component<Props> {
...
-        fetchers={[
-          createErc20DividendsModuleBySymbolFetcher({ securityTokenSymbol }),
-        ]}
-        render={(data: {
-          erc20DividendsModules: types.Erc20DividendsModuleEntity[];
-        }) => {
-          const {
-            erc20DividendsModules: [erc20DividendsModule],
-          } = data;
+    // return (
```
[the remark this answers is not in the document]
Comment (Grsmto) [reply packages/new-polymath-issuer/src/pages/SecurityTokensDividends/Container.tsx]: Seems to work fine now 👍

File: packages/new-polymath-issuer/src/components/Header/Container.tsx [hunk 9]
```diff
@@ -7,16 +7,29 @@ import { RootState } from '~/state/store';
...
 
-export interface Props extends HeaderProps {
-  walletAddress: string;
-}
+export interface Props
+  extends Pick<HeaderProps, 'variant' | 'walletAddress' | 'RouterLink'> {}
 
-const mapStateToProps = (state: RootState): Props => ({
+const mapStateToProps = (state: RootState) => ({
   walletAddress: get(state, 'session.wallet.address'),
```
[the remark this answers is not in the document]
Comment (Grsmto) [reply packages/new-polymath-issuer/src/components/Header/Container.tsx]: Yeah. But in this case there is no computation happening so Reselect would be useless. It's just getting the value from a JS object. Right?

File: packages/new-polymath-issuer/src/components/enhancers/DataFetcher.tsx [hunk 10]
```diff
@@ -69,10 +60,39 @@ class DataFetcherBase extends Component<Props> {
...
+  public componentDidUpdate() {
     this.getData();
   }
 
+  /**
+   * Only update if the fetchers change
+   */
+  public shouldComponentUpdate(
+    nextProps: Readonly<Props>,
+    _nextState: Readonly<{}>
```
Comment (RafaelVidaurre) [inline packages/new-polymath-issuer/src/components/enhancers/DataFetcher.tsx]: Can we remove this? Or is it necessary for some typing reasons? @monitz87

Comment (RafaelVidaurre) [reply packages/new-polymath-issuer/src/components/enhancers/DataFetcher.tsx]: @monitz87
